core/http_bench.rs

```diff
@@ -29,62 +28,93 @@ const OP_READ: i32 = 3;
 const OP_WRITE: i32 = 4;
 const OP_CLOSE: i32 = 5;
 
-const INDEX_START: usize = 0;
-const INDEX_END: usize = 1;
-
-const NUM_RECORDS: usize = 128;
-const RECORD_SIZE: usize = 4;
-
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct Record {
   pub promise_id: i32,
   pub op_id: i32,
   pub arg: i32,
   pub result: i32,
 }
 
-pub type HttpBenchOp = dyn Future<Item = i32, Error = std::io::Error> + Send;
+impl Into<Buf> for Record {
+  fn into(self) -> Buf {
+    let buf32 = vec![self.promise_id, self.op_id, self.arg, self.result]
+      .into_boxed_slice();
+    let ptr = Box::into_raw(buf32) as *mut [u8; 16];
+    unsafe { Box::from_raw(ptr) }
+  }
+}
 
-struct HttpBench {
-  shared32: Vec<i32>,
+impl From<&[u8]> for Record {
+  fn from(s: &[u8]) -> Record {
+    let ptr = s.as_ptr() as *const i32;
+    let ints = unsafe { std::slice::from_raw_parts(ptr, 4) };
+    Record {
+      promise_id: ints[0],
+      op_id: ints[1],
+      arg: ints[2],
+      result: ints[3],
+    }
+  }
 }
 
-impl HttpBench {
-  fn new() -> Self {
-    let mut shared32 = Vec::<i32>::new();
-    let n = 2 + 4 * NUM_RECORDS;
-    shared32.resize(n, 0);
-    shared32[INDEX_START] = 0;
-    shared32[INDEX_END] = 0;
-    Self { shared32 }
+impl From<Buf> for Record {
+  fn from(buf: Buf) -> Record {
+    assert_eq!(buf.len(), 4 * 4);
+    //let byte_len = buf.len();
+    let ptr = Box::into_raw(buf) as *mut [i32; 4];
+    let ints: Box<[i32]> = unsafe { Box::from_raw(ptr) };
+    assert_eq!(ints.len(), 4);
+    Record {
+      promise_id: ints[0],
+      op_id: ints[1],
+      arg: ints[2],
+      result: ints[3],
+    }
   }
 }
 
-fn idx(i: usize, off: usize) -> usize {
-  2 + i * RECORD_SIZE + off
+#[test]
+fn test_record_from() {
+  let r = Record {
+    promise_id: 1,
+    op_id: 2,
+    arg: 3,
+    result: 4,
+  };
+  let expected = r.clone();
+  let buf: Buf = r.into();
+  // TODO How do only do this on little endian?
```

Comment:
You mean big endian?
All platforms are little endian except some exotic ones.

Reply:
er ya meant "How to do this only on little endian"

Reply:
`[cfg(target_endian = "little")]`

Reply:
thx!